core/src/main/java/jenkins/util/io/PathRemover.java

```diff
@@ -209,26 +226,30 @@ public String failureMessage(@NonNull Path fileToRemove, int retryCount) {
         }
     }
 
-    private List<IOException> tryRemoveRecursive(@NonNull Path path) {
+    private void tryRemoveRecursive(@NonNull Path path, List<IOException> accumulatedErrors) {
```

Comment:
I modified this method and `tryRemoveDirectoryContents` to accept the list of errors as a parameter and mutate it instead of creating new lists at every level so that `shouldAbort` can act on the number of exceptions seen at every level rather than only those seen in the current directory.